costmodel/router.go

```diff
@@ -74,14 +74,24 @@ type DataEnvelope struct {
 	Message string      `json:"message,omitempty"`
 }
 
-// FilterCostData allows through only CostData that matches the given filters for namespace and clusterId
-func FilterCostData(data map[string]*CostData, namespace, clusterId string) map[string]*CostData {
+// FilterFunc is a filter that returns true iff the given CostData should be filtered out
+type FilterFunc func(*CostData) bool
+
+// FilterCostData allows through only CostData that matches all the given filter functions
+func FilterCostData(data map[string]*CostData, filters ...FilterFunc) map[string]*CostData {
 	result := make(map[string]*CostData)
+
+DataLoop:
```

Comment:
nit: is everything here supposed to be tabbed one more indent out?

Reply:
I thought the same thing, but gofmt corrects it to this formatting.

Reply:
okay LGMT